docs/language/functions/regexp_replace.md

```diff
@@ -10,28 +10,50 @@ regexp_replace(s: string, re: string|regexp, new: string) -> string
 ### Description
 
 The _regexp_replace_ function substitutes all characters matching the
-[regular expression](../overview.md#regular-expressions) `re` in string `s` with
+[regular expression](../overview.md#811-regular-expressions) `re` in string `s` with
```

Comment:
Do you know why .github/workflows/markdown-lint.yml didn't catch as part of #4435?

Reply:
@nwt: Well, I'm not surprised by it... does that count? 😄  I first confronted this limitation in #4028. While the [markdown link checker](https://github.com/gaurav-nelson/github-action-markdown-link-check) we use can detect broken anchor links within the same doc, it's never been smart enough to detect them when pointing to other docs. I think this is partially GitHub's fault because if you follow the link you get an HTTP 200 and it puts you at the top of the doc and sends back an HTTP 200. Here's that with the broken link of the moment:

```
$ curl -s -o /dev/null -w "%{http_code}" "https://github.com/brimdata/zed/blob/main/docs/language/overview.md#regular-expressions"
200
```

Meanwhile the [link checker we use on deployed sites](https://github.com/ruzickap/action-my-broken-link-checker) has always been able to catch these. I can't say why all the way down to the code level since the behavior & response code seem much the same on the rendered docs site.

```
$ curl -s -o /dev/null -w "%{http_code}" "https://zed.brimdata.io/docs/next/language/overview#regular-expressions"
200
```

However, that link checker is somehow still able to see this as a breakage, as shown in the output of a [recent zed-docs-site Actions run](https://github.com/brimdata/zed-docs-site/actions/runs/4483517347/jobs/7882899690):

```
...
2023-03-21T20:31:56.3361518Z https://zed.brimdata.io/docs/next/language/functions/regexp_replace
...
2023-03-21T20:31:56.3381315Z    id #regular-expressions not found       https://zed.brimdata.io/docs/next/language/overview#regular-expressions
...
```

If I had to guess, I think the [muffet](https://github.com/raviqqe/muffet) tool that's used by the checker that's catching the breakage is more sophisticated, whereas the markdown link checker depends on https://github.com/tcort/markdown-link-check which looks simpler and has a bunch of confusing open/closed issues grumping about problems with anchors, and as far as I can tell they've decided not to cover this case. I can't remember if at one time I explored phasing out the markdown-level one altogether but I seem to recall that treating `https://github.com/brimdata/zed/tree/main/docs` like a rendered site caused its own set of problems (HTTP 429, maybe?). So long ago I resolved that it was ok to have a "belt & suspenders" approach where the markdown one caught _most_ of the breakages early and then the second link checker was there just in case any snuck by.